src/window.rs

```diff
@@ -37,29 +40,27 @@ use zeroize::Zeroize;
 /// only `pub(crate)` so that we can write hardcoded constants, so it's
 /// still technically possible.  It would be nice to prevent direct
 /// access to the table.
-///
-/// XXX make this generic with respect to table size
 #[derive(Copy, Clone)]
-pub struct LookupTable<T>(pub(crate) [T; 8]);
+pub struct $name<T>(pub(crate) [T; $size]);
 
-impl<T> LookupTable<T>
+impl<T> $name<T>
 where
     T: Identity + ConditionallySelectable + ConditionallyNegatable,
 {
     /// Given \\(-8 \leq x \leq 8\\), return \\(xP\\) in constant time.
     pub fn select(&self, x: i8) -> T {
-        debug_assert!(x >= -8);
-        debug_assert!(x <= 8);
+        debug_assert!(x >= $neg);
+        debug_assert!(x as i16 <= $size as i16); // XXX We have to convert to i16s here for the radix-256 case.. this is wrong.
```

Comment:
I can see why this is required ($size == 258 in radix-256, whereas i8::MAX == 257)... but why is the conversion to i16 "wrong"? Will it result in incorrect calculations (it doesn't seem so), or is it just bad practice?

Would it be better to make x an i16 to start with, or would that incur too much of a performance cost?

Reply:
It's wrong because `$size` is 128 in radix-256, but `x` is an `i8` which has a max of 127, so `x` must be less than or equal to `$size` by virtue of the type system. We can't make it an `i16` without also changing the return type of `Scalar.to_radix_2w()`, because the bytes returned from the latter are what are used to call `select()` here.

Reply:
Ah, also it's even more wrong, because in debug mode this should panic when we try to `select(128)`, but the test basepoints and scalars I wrote are static and don't seem to trigger this. I should probably write a fuzzer and see if I can get it to panic.

Reply:
Okay, I see how this is tricky/wrong for radix-256. 

Would you like to address those issues (making sure it panics when you call `select(128)`, writing fuzzers, etc) in this PR? Or, I can LGTM this PR with your placeholder comment, so you can address that in a later PR / so it doesn't block anything else.

Reply:
I think I'd like fuzzers and better tests for all the sizes, even though I'm pretty thoroughly convinced that everything less than radix-256 is correct. I can try to get to it this evening.